### PR TITLE
Add token argument

### LINKED ISSUE #240: Add a token argument to otter assign
To bypass using username and password for gradescope, it would be helpful to be able to just specify your token

### PR DESCRIPTION
**Pull request by JinwooPark00**
Closes #240

Feature Implementation
- Implemented token as an argument for Otter Generate, which can be used to bypass the need for username and password entry.
- Modified Otter Assign to run Otter Generate with optional token from the assignment meta data.

Testing
- Added a test asserting that autograder.zip files generated with token has same contents as the 'correct' directory with updated config file containing token (test generate).
- Updated metadata in generate-gradescope.ipynb to include token, updated gs-correct accordingly to ensure tests in test_assign.py test properly with the changes.

### CODE REVIEW

File: otter/assign/utils.py [hunk 1]
```diff
@@ -227,6 +227,7 @@ def run_generate_autograder(result, assignment, gs_username, gs_password, plugin
         # generate_args['token'] = token
         generate_args['course_id'] = str(pdf_args['course_id'])
         generate_args['assignment_id'] = str(pdf_args['assignment_id'])
+        generate_args['token'] = str(pdf_args['token'])
```
Comment (chrispyles) [inline otter/assign/utils.py:230]: Cannot assume that the `token` key will be provided. The default behavior should still be available if, e.g., the user does something like:

```yaml
generate:
    pdfs:
        course_id: 123456
        assignment_id: 456789
```

**Review by chrispyles — CHANGES_REQUESTED** [review]
Left one comment. Please fix and we can take another look.

**Review by chrispyles — CHANGES_REQUESTED** [review]
Also, I notice you didn't modify `docs/otter_assign/python_notebook_format.md`. Please make sure to add something like [this](https://github.com/ucbds-infra/otter-grader/pull/253/files#diff-6e41a5618b9d7266f9c116a6db60fd02276860703a6296b1fd1c1b48bed01e8eR49).

File: otter/assign/utils.py (hunk 1 above)
Comment (JinwooPark00) [reply otter/assign/utils.py]: updated, not sure why but .get() only seems to work if colon is included as part of the key value

**Review by chrispyles — CHANGES_REQUESTED** [review]
Make sure to add this file to the list in release.py

test/test_generate/test-autograder/autograder-token-correct/requirements.txt

**Comment by chrispyles** [comment]
@JinwooPark00 I've taken your branch and added some commits to it so that it can be merged into the beta of v3 instead of into master. I'm going to close this PR in favor of #285 which includes all the commits from this one but merges into `beta`.

**Pull request closed without merging**